src/main/java/com/mongodb/hibernate/internal/translate/mongoast/command/aggregate/stage/AstProjectStageSpecification.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2025-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.hibernate.internal.translate.mongoast.command.aggregate.stage;
+
+import com.mongodb.hibernate.internal.translate.mongoast.AstNode;
+import org.bson.BsonWriter;
+
+public abstract class AstProjectStageSpecification implements AstNode {
```

Comment:
Let's introduce a private constructor, thus making it possible to extend `AstProjectStageSpecification` only from the same compilation unit the abstract class is declared in (the `AstProjectStageSpecification.java` file), thus limiting its extensibility.

```suggestion
public abstract class AstProjectStageSpecification implements AstNode {
    private AstProjectStageSpecification() {
    }
```

Reply:
adopted. The `stage` package has been removed, so I could not 'commit suggestion' directly.